interp/interpreter.go

```diff
@@ -736,30 +736,25 @@ func (r *runner) run(fn *function, params []value, parentMem *memoryView, indent
 			if err == nil {
 				// The lhs is a pointer. This sometimes happens for particular
 				// pointer tricks.
-				switch inst.opcode {
-				case llvm.Add:
+				if inst.opcode == llvm.Add {
```

Comment:
This looks like a good place for:

```go
switch {
    case inst.opcode == llvm.Add:
        // do stuff
    case inst.opcode == llvm.Xor && rhs.Uint() == 0:
        // other stuff
    default:
        // yet other stuff
}
```

Reply:
I just noticed that it is not quite so easy to replace that code in this PR without changing how the commits are structured.

If I feel strongly I will submit a PR with that change.

Reply:
Oh, yes that's another option. Didn't think of that. It has the same effect though.

I don't really care about the specific form 🤷 I can update the PR if you like.